lib/include/openamp/rpmsg_virtio.h

```diff
@@ -27,7 +27,8 @@ extern "C" {
 #endif
 
 /* The feature bitmap for virtio rpmsg */
-#define VIRTIO_RPMSG_F_NS 0 /* RP supports name service notifications */
+#define VIRTIO_RPMSG_F_NS	0 /* RP supports name service notifications */
+#define VIRTIO_RPMSG_F_BUFSZ	2 /* RP supports get buffer size from config space */
```

Comment:
Very Interesting feature. I would suggest you to propose this in a first step in Linux, to have a common implementation in  Linux and OpenAMP. If someone push another implementation on Linux side, this could not be compatible...

Reply:
Yes, I am preparing Linux side patch. once OpenAMP get approved, we will submit the similar change to kernel side too.

Reply:
let's Wendy decide the best strategy. By experience i would suggest to push first or in parallel in Linux...

Reply:
Sure, once the review is good here, I will submit the patch to Linux kernel ASAP.

Reply:
For the master sending message to slave, the buffer size ideally should depends on the actually message size. It looks like this flag is because the master also allocates buffers for slave to send messages, and the message for the slave to send can be larger than the fixed size defined by the master. Have a discussion with @andersson, this change negotiate the size before vrings setup. Will need more time to consider if this is an appropriate change.

Reply:
@andersson, the kernel side hard code rpmsg buffer size to MAX_RPMSG_BUF_SIZE(512), it isn't always best choice for the remote side:
1.For small MCU, we want to reduce the buffer size to 256 or even 128
2.But for NBIoT(WiFi), we want to extend to MTU(about 1500)
My proposal is:
1.Add new rx/tx buffer size field into config space
2.Master allocate and split the buffer based on these value
So firmware writer could choice:
1.The best rpmsg buffer number from fw_rsc_vdev_vring->num
2.The best rpmsg buffer size from virtio_rpmsg_config->[rx|tx]buf_size
Of course, to keep the backward compatibility, all new behaviour is guarded by feature bit(VIRTIO_RPMSG_F_BUFSZ).
Do you prefer I submit a kernel patch for discussion?

Reply:
> Do you prefer I submit a kernel patch for discussion?

Yes, please, as this is an interesting feature, if you post it to the Linux kernel, more people will be involved in the discussion. Thanks, Wendy